// llama-parse/dist/index.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var __async = (__this, __arguments, generator) => {
  return new Promise((resolve, reject) => {
    var fulfilled = (value) => {
      try {
        step(generator.next(value));
      } catch (e) {
        reject(e);
      }
    };
    var rejected = (value) => {
      try {
        step(generator.throw(value));
      } catch (e) {
        reject(e);
      }
    };
    var step = (x) => x.done ? resolve(x.value) : Promise.resolve(x.value).then(fulfilled, rejected);
    step((generator = generator.apply(__this, __arguments)).next());
  });
};

// src/index.ts
var index_exports = {};
__export(index_exports, {
  LlamaParse: () => LlamaParse
});
module.exports = __toCommonJS(index_exports);
var LlamaParse = class {
  /**
   * Create a new LlamaParse instance
   * @param config Configuration options for LlamaParse
   * @throws {Error} If API key is not provided
   */
  constructor(config) {
    if (!config.apiKey) throw new Error("API key is required");
    this.baseUrl = config.baseUrl || "https://api.cloud.llamaindex.ai/api/parsing";
    this.headers = __spreadValues({
      Authorization: `Bearer ${config.apiKey}`,
      accept: "application/json"
    }, config.headers);
  }
  /**
   * Parse a PDF file and convert it to markdown
   * @param file File or Blob to parse
   * @returns Promise containing the markdown result and metadata
   */
  parseFile(file) {
    return __async(this, null, function* () {
      const jobId = yield this.uploadFile(file);
      let status;
      do {
        status = yield this.checkStatus(jobId);
        yield new Promise((resolve) => setTimeout(resolve, 1e3));
      } while (status.status === "PENDING");
      if (status.status !== "SUCCESS") {
        throw new Error(`Parsing failed: ${status.error}`);
      }
      return this.getMarkdownResult(jobId);
    });
  }
  /**
   * Upload a PDF file to start the parsing process
   * @param file File or Blob to upload
   * @returns Promise containing the job ID
   * @throws {Error} If upload fails
   */
  uploadFile(file) {
    return __async(this, null, function* () {
      const formData = new FormData();
      formData.append("file", file);
      const response = yield fetch(`${this.baseUrl}/upload`, {
        method: "POST",
        headers: this.headers,
        body: formData
      });
      if (!response.ok) {
        throw new Error(`Upload failed: ${response.statusText}`);
      }
      const data = yield response.json();
      return data.id;
    });
  }
  /**
   * Check the status of a parsing job
   * @param jobId ID of the job to check
   * @returns Promise containing the job status
   * @throws {Error} If status check fails
   */
  checkStatus(jobId) {
    return __async(this, null, function* () {
      const response = yield fetch(`${this.baseUrl}/job/${jobId}`, {
        headers: this.headers
      });
      if (!response.ok) {
        throw new Error(`Status check failed: ${response.statusText}`);
      }
      return response.json();
    });
  }
  /**
   * Get the markdown result for a completed job
   * @param jobId ID of the completed job
   * @returns Promise containing the markdown result
   * @throws {Error} If retrieval fails
   */
  getMarkdownResult(jobId) {
    return __async(this, null, function* () {
      const response = yield fetch(`${this.baseUrl}/job/${jobId}/result/markdown`, {
        headers: this.headers
      });
      if (!response.ok) {
        throw new Error(`Failed to get results: ${response.statusText}`);
      }
      return response.json();
    });
  }
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  LlamaParse
});
